src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int NO_INDEX = -1;
+    private static final double DEFAULT_LOAD_FACTOR = 1.5;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayIsFull();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Requested index " + index + " is out of bounds");
+        } else {
+            growIfArrayIsFull();
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+            elementData[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] buffer = toArray(list);
+        for (Object element : buffer) {
+            add((T) element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexOutOfSize(index);
```

Comment:
```suggestion
        checkIndex(index);
```

prefix 'is' is usually used when method return boolean